test/src/e2e.dynval/dv.n.test.ts

```diff
@@ -0,0 +1,304 @@
+            0x0200, // maxTextContentSize
+            "tc", // networkID
+            10, // minPayCost
+            10, // minSetRegularKeyCost
+            10, // minCreateShardCost
+            10, // minSetShardOwnersCost
+            10, // minSetShardUsersCost
+            10, // minWrapCccCost
+            10, // minCustomCost
+            10, // minStoreCost
+            10, // minRemoveCost
+            10, // minMintAssetCost
+            10, // minTransferAssetCost
+            10, // minChangeAssetSchemeCost
+            10, // minIncreaseAssetSupplyCost
+            10, // minComposeAssetCost
+            10, // minDecomposeAssetCost
+            10, // minUnwrapCccCost
+            4194304, // maxBodySize
+            16384, // snapshotPeriod
+            TERM_SECONDS, // termSeconds
+            10, // nominationExpiration
+            10, // custodyPeriod
+            30, // releasePeriod
+            8, // maxNumOfValidators
+            4, // minNumOfValidators
+            1000, // delegationThreshold
+            10000, // minDeposit
+            128 // maxCandidateMetadataSize
+        ];
+        const changeParams: [
+            number,
+            number,
+            (number | string)[],
+            ...string[]
+        ] = [0xff, 0, newParams];
+        const message = blake256(RLP.encode(changeParams).toString("hex"));
+        changeParams.push(`0x${SDK.util.signEcdsa(message, faucetSecret)}`);
+        changeParams.push(`0x${SDK.util.signEcdsa(message, aliceSecret)}`);
+        changeParams.push(`0x${SDK.util.signEcdsa(message, bobSecret)}`);
+
+        return await node.sdk.rpc.chain.sendSignedTransaction(
+            node.sdk.core
+                .createCustomTransaction({
+                    handlerId: stakeActionHandlerId,
+                    bytes: RLP.encode(changeParams)
+                })
+                .sign({
+                    secret: faucetSecret,
+                    seq: await node.sdk.rpc.chain.getSeq(faucetAddress),
+                    fee: 10
+                })
+        );
+    }
+
+    async function waitForTermChange(target: number, timeout?: number) {
+        const start = Date.now();
+        while (true) {
+            const termMetadata = await stake.getTermMetadata(nodes[0].sdk);
+            if (termMetadata && termMetadata.currentTermId === target) {
+                break;
+            }
+            await wait(1000);
+            if (timeout) {
+                if (Date.now() - start > timeout * 1000) {
+                    throw new Error(`Term didn't changed in ${timeout} s`);
+                }
+            }
+        }
+    }
+
+    beforeEach(async function() {
+        await bootstrap();
+    });
+
+    describe("1. No delegation, nominate, revoke, jail", function() {
+        it("should switchs term within designated term seconds", async function() {
+            const margin = 1.1;
+
+            await waitForTermChange(1, TERM_SECONDS * margin);
```

Comment:
Here you waited until the term changes to 1 and just after you are expecting the `currentTermId` to be equal to 2. I didn't catch your intention. Can you explain it?

Reply:
term 0 -> term 1 is ruled by validators specified in the scheme (static validators period). 
At the first `termEnd`, term 1 starts and validators are changed to dynamic validators. `await bootstrap()` makes sure test cases are in the dynamic validator period. 
The test case checks that "term change occurs in the dynamic validator period"